### DISCUSSION TITLE
Problem with making changes to styles. (JS)

### DISCUSSION THREAD

**Discussion by ZigglerPL** [community/community#120229] [Programming Help]
### Body

Hello, what could have happened that the CSS rules defined in the animation object are not assigned to the element at all after the animation is completed?

I don't have any visible errors in the console and the code structure seems logical to me.

Suddenly it stopped working globally for any element.

[https://playcode.io/httpstestowaniekodu](url)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by GunaPalanivel** [comment]
Hello,

It sounds like you're experiencing a tricky issue where the CSS properties defined in your animation keyframes aren't being retained by the elements once the animation has completed. Here are a few things you might consider checking and adjusting to resolve this issue:

1. **Check the `fill` Property**: The behavior you're describing can often be attributed to the absence of the `fill` property in your animation parameters. By default, the effects of an animation only apply for the duration of the animation. If you want the final state of the animation (like the scale and background color at the last keyframe) to persist after the animation ends, you should set the `fill` mode to `'forwards'`. This mode ensures that the CSS applied by the final keyframe of the animation will persist after the animation has finished. You can adjust your `parametersAnimation` object like this:

    ```javascript
    let parametersAnimation = {
        duration: numberDuration,
        iterations: numberIterations,
        fill: 'forwards', // Ensures final state of animation persists
        easing: 'ease'
    };
    ```

2. **Verify Keyframes and Styles**: Double-check your keyframes to ensure they are correctly defined and that the styles applied at each stage of the animation are what you intend. Sometimes, a small typo or logical error can disrupt the entire flow of the animation.

3. **Browser Compatibility**: Ensure that the animation features you're using are supported in the browser where you're testing your code. While modern browsers support CSS animations well, there could be differences or bugs that affect how animations behave.

4. **Check for Conflicting CSS**: Look for any CSS rules that might be overriding your animation effects once the animation completes. This can particularly be a case when using CSS frameworks or additional stylesheets that might have higher specificity or !important declarations.

5. **Inspect Element State**: Use your browser's developer tools to inspect the animated elements both during and after the animation. Check the computed style values to see if they match what's expected based on your keyframes. This can provide clues as to why the expected styles aren't being retained.

6. **Console Errors**: Since you mentioned there are no visible errors in the console, it might still be helpful to add more `console.log` statements around your animation code to ensure all parts of your function are executing as expected.

If after trying these steps your issue still isn't resolved, it might help to isolate the problem further by creating a minimal reproduction of the issue in a separate clean environment, which could help identify whether external factors are affecting the animation.

Best of luck, and I hope this helps! If you need further assistance, feel free to reach out again.
